frontend/javascripts/oxalis/controller/url_manager.js

```diff
@@ -103,28 +131,33 @@ class UrlManager {
     window.onhashchange = () => this.onHashChange();
   }
 
-  buildHash(tracing: Tracing) {
-    const position = V3.floor(getPosition(Store.getState().flycam));
-    const { viewMode } = Store.getState().temporaryConfiguration;
-    const viewModeIndex = ViewModeValues.indexOf(viewMode);
-    const zoomStep = Store.getState().flycam.zoomStep.toFixed(3);
-    const rotation = constants.MODES_ARBITRARY.includes(viewMode)
-      ? getRotation(Store.getState().flycam).map(e => e.toFixed(2))
-      : [];
+  buildUrlHash(state: OxalisState) {
+    const position = V3.floor(getPosition(state.flycam));
+    const { viewMode: mode } = state.temporaryConfiguration;
+    const zoomStep = Utils.roundTo(state.flycam.zoomStep, 3);
+    const rotationOptional = constants.MODES_ARBITRARY.includes(mode)
+      ? { rotation: getRotation(state.flycam).map(e => Utils.roundTo(e, 2)) }
+      : {};
 
-    const activeNodeId = getSkeletonTracing(tracing)
+    const activeNodeOptional = getSkeletonTracing(state.tracing)
       .chain(skeletonTracing => getActiveNode(skeletonTracing))
-      .map(node => [node.id])
-      .getOrElse([]);
+      .map(node => ({ activeNode: node.id }))
+      .getOrElse({});
+
+    // $FlowIssue[exponential-spread] See https://github.com/facebook/flow/issues/8299
+    const urlState = { position, mode, zoomStep, ...rotationOptional, ...activeNodeOptional };
 
-    return [...position, viewModeIndex, zoomStep, ...rotation, ...activeNodeId].join(",");
```

Comment:
From our verbal discussion: The legacy variant (maybe we should call it `CSV` since we'll stick to it) should be still supported and also used here  for updating the URL (probably the component would need to support both ways for building the hash; the new variant would be used by the share modal).